### PR TITLE
feat: add chunk split plugin in webpack-builder

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

增加 webpack-builder 拆包功能。

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/webpack-builder/src/plugins/splitChunks.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  return cacheGroups;
+}
+
+function splitByExperience(ctx: SplitChunksContext): SplitChunks {
+  const { override, userDefinedCacheGroups, defaultConfig } = ctx;
+  const experienceCacheGroup: CacheGroup = {};
+  const SPLIT_EXPERIENCE_LIST = [
+    /[\\/]react|react-dom[\\/]/,
+    /[\\/]react-router|react-router-dom|history[\\/]/,
+    /[\\/]@ies\/ace[\\/]/,
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/splitChunks.ts:57]: list 里不应该直接出现 @ies/ace 这样的内部库，能否外置到其他地方

File: packages/builder/webpack-builder/src/plugins/splitChunks.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    cacheGroups: {
+      ...defaultConfig.cacheGroups,
+      ...userDefinedCacheGroups,
+      // Core group
+      vendors: {
+        priority: -10,
+        test: /[\\/]node_modules[\\/]/,
+        name(module: Module): string | false {
+          const handleModuleContext = module?.context?.match(
+            /[\\/]node_modules[\\/](\.pnpm[\\/])?(?:(@[^[\\/]+)(?:[\\/]))?([^\\/]+)/,
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/splitChunks.ts:110]: 这个正则有点复杂，可以把这个正则的逻辑抽取成一个函数，然后补充几个相关的单测

File: packages/builder/webpack-builder/src/plugins/splitChunks.ts (hunk 1 above)
Comment (sanyuan0704) [reply packages/builder/webpack-builder/src/plugins/splitChunks.ts:57]: 这个还是交给用户去配置吧

Comment (chenjiahan) [reply packages/builder/webpack-builder/src/plugins/splitChunks.ts:57]: 嗯呐，理想情况下，webpack-builder 里的逻辑要尽可能通用，xgplayer 这种外部不太常用的库也不推荐写死在里面

Comment (sanyuan0704) [reply packages/builder/webpack-builder/src/plugins/splitChunks.ts:57]: OK

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
🎉🎉

**Pull request merged**